### DISCUSSION TITLE
Custom widget: How to add_callbacks "Button1": self.poll as default?

### DISCUSSION THREAD

**Discussion by thingsiplay** [qtile/qtile#3712] [Q&A] [answered]
I am trying to create my first Qtile widget and can't figure out one thing since yesterday. It is based on ThreadPoolText and is very simple. The poll() method works as intended automatically updating the widget, let's say every 5 seconds. I want to add a default callback to that runs poll() whenever I click with left mouse click. Like this:

    self.add_callbacks(
        {
            "Button1": self.poll,
            # "Button1": lazy.spawn("kitty"),
            # "Button1": self._update(),
        }
    )

Note, I tried lazy.spawn() and that works, but not poll. On a sidenote, configuring widget on mouse_callbacks at user config.py level works too, like this: `lazy.widget['checkmails'].eval('self.update(self.poll())'),` but I have to give it a name beforehand. What am I missing here? How can I just run self.poll? I tried something lazy too:

    @lazy.function
    def _update(self):
        self.poll()

But won't work.

**Comment by elParaguayo** [comment, marked as answer]
I don't think you want `self.poll`. That just returns the text to update the widget.

What if you bind `self.cmd_force_update`?

**Reply by thingsiplay** [reply]
Oh, you are right. Off course it could not worked like that. Thank you `self.cmd_force_update` works and updates the widget.
